charts/kong/templates/custom-resource-definitions.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.ingressController.enabled .Values.ingressController.installCRDs -}}
+{{- if (or (and .Values.ingressController.enabled .Values.ingressController.installCRDs) (and (not (or .Values.deployment.kong.enabled .Values.ingressController.enabled )) .Values.ingressController.installCRDs)) -}}
```

Comment:
Please add a comment in the code to explain this horrendous if condition and why we are doing that too. It will be very easy to forget the why part in future.

Reply:
I'm myself having a hard time understanding this.